addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -257,17 +260,19 @@ const threadPatch = {
     get showUnreadBanner() {
         return this.self_member_id?.message_unread_counter_ui > 0;
     },
-    executeCommand(command, body = "") {
-        command.onExecute?.(this.channel);
-        return this.store.env.services.orm.call(
-            "discuss.channel",
-            command.methodName,
-            [[this.id]],
-            { body }
-        );
+    async executeCommand(command, body = "") {
+        await command.onExecute?.(this.channel);
+        if (command.methodName) {
+            await this.store.env.services.orm.call(
+                "discuss.channel",
+                command.methodName,
+                [[this.id]],
+                { body }
+            );
+        }
     },
     async leaveChannel() {
-        if (this.channel?.channel_type !== "group" && this.create_uid?.eq(this.store.self_user)) {
+        if (this.channel?.channel_type === "channel" && this.create_uid?.eq(this.store.self_user)) {
```

Comment:
Note: this condition is probably obsolete with channel_role and should be adapted in a follow up.

Regardless, having this check done in chat, livechat and other types made no sense.